### PR TITLE
create lint-go script and call it from Makefile

### PR DESCRIPTION
**Pull request by AldoFusterTurpin**
### Which issue this PR addresses:

Fixes vendored linter not working locally

### What this PR does / why we need it:

Previous to this PR, when executing `make test-go` or `make lint-go` makefile was invoking vendored golangci-lint and it returned some lint errors as it's an old linter version, thus breaking the local development workflow.  

With the proposed changes, when running any of the above commands:
- if golangci-lint binary is present in dev machine, it will run the local golangci-lint binary
- if golangci-lint binary is NOT present in dev machine, it will just show a warning about skipping linting and the link to download golangci-lint

### Test plan for issue:
Test it locally when golangci-lint exist and when does not exist.

### Is there any documentation that needs to be updated for this PR?

No, simply fix dev workflow.

### CODE REVIEW

File: hack/lint-go.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+if ! command -v golangci-lint &> /dev/null
+then
+    echo "WARNING golangci-lint could not be found, golang lint skipped.To install it: https://golangci-lint.run/usage/install/#local-installation"
```
Comment (facchettos) [inline hack/lint-go.sh:3]: I would probably fail the script if we can't find it 
```suggestion
    echo "WARNING golangci-lint could not be found, golang lint skipped.To install it: https://golangci-lint.run/usage/install/#local-installation"
    exit 1
```

Comment (s-amann) [reply hack/lint-go.sh:3]: +1 to failing the script as it more strongly encourages installing the linter

**Review by s-amann — DISMISSED** [review]

File: hack/lint-go.sh (hunk 1 above)
Comment (AldoFusterTurpin) [reply hack/lint-go.sh:3]: Thank you both for the feedback 🙂
 Done!

**Review by s-amann — APPROVED** [review]
LGTM

**Review by cadenmarchese — COMMENTED** [review]
Thanks, this has been bugging me as well lately.

I think you could get away with having the script inside of the Makefile itself, I am not sure a separate file is needed.

If you'd rather keep it in a separate file, maybe add a [shebang and code comment](https://github.com/Azure/ARO-RP/blob/master/hack/ssh-agent.sh#L1-L5) to align it with our other hack scripts.

Thanks!

**Comment by AldoFusterTurpin** [comment]
@cadenmarchese Great idea. Thanks! 
Done.

**Review by petrkotas — APPROVED** [review]
Thanks, good job @AldoFusterTurpin

**Review by s-amann — DISMISSED** [review]

**Review by darthhexx — DISMISSED** [review]
LGTM. Small nit about a \n.

File: hack/lint-go.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# In case the binary that is trying to execute does not exist,
+# the script will print a message with instructions to download the binary and will exit with a non-zero exit code.
+
+if ! command -v golangci-lint &> /dev/null
+then
+    echo "ERROR: golangci-lint could not be found, Go linting aborted. To install it visit https://golangci-lint.run/usage/install/#local-installation"
+    exit 1
+else
+    golangci-lint run
+fi
```
[the remark this answers is not in the document]
Comment (AldoFusterTurpin) [reply hack/lint-go.sh]: Thank you. Done

**Review by petrkotas — APPROVED** [review]
Thanks @AldoFusterTurpin for this improvement!

**Pull request merged**